cmd/lint.go

```diff
@@ -91,7 +95,102 @@ func LintCmd(lg log.Logger, mt *metrics.StatsdMonitor) *cobra.Command {
 			}
 			fmt.Printf("%d failing, %d successful, and %d total\n\n", failures, success, len(recipes))
 			printer.Table(os.Stdout, report)
+
 			return nil
 		},
 	}
 }
+
+// printLintErrors prints the recipe errors
+func printLintErrors(errs []error, rcp recipe.Recipe) {
+	var notFoundError plugins.NotFoundError
+	var invalidConfigError plugins.InvalidConfigError
+	for _, err := range errs {
+		if errors.As(err, &notFoundError) {
+			printPluginError(rcp, notFoundError)
+			continue
+		}
+		if errors.As(err, &invalidConfigError) {
+			printConfigError(rcp, invalidConfigError)
+			continue
+		}
+		fmt.Printf("recipe error: %s", err.Error())
+	}
+}
+
+// printPluginError print the plugin type error
+func printPluginError(rcp recipe.Recipe, notFoundError plugins.NotFoundError) {
+	if notFoundError.Type == plugins.PluginTypeExtractor {
+		line := rcp.Source.Node.Name.Line
+		fmt.Printf("%s: invalid extractor on line: %d\n", rcp.Name, line)
+	} else if notFoundError.Type == plugins.PluginTypeProcessor {
+		plugin, exists := findPluginByName(rcp.Processors, notFoundError.Name)
+		if exists {
+			line := plugin.Node.Name.Line
+			fmt.Printf("%s: invalid processor on line: %d\n", rcp.Name, line)
+		}
+	} else if notFoundError.Type == plugins.PluginTypeSink {
+		plugin, exists := findPluginByName(rcp.Sinks, notFoundError.Name)
+		if exists {
+			line := plugin.Node.Name.Line
+			fmt.Printf("%s: invalid sink on line: %d\n", rcp.Name, line)
+		}
+	}
+}
+
+// printConfigError print the plugin config error
+func printConfigError(rcp recipe.Recipe, invalidConfigError plugins.InvalidConfigError) {
+	if invalidConfigError.Type == plugins.PluginTypeExtractor {
+		pluginNode := rcp.Source.Node
+		for _, configError := range invalidConfigError.Errors {
+			cfg, ok := pluginNode.Config[configError.Key]
+			if ok {
+				line := cfg.Line
+				fmt.Printf("%s: invalid %s extractor config on line: %d\n", rcp.Name, invalidConfigError.PluginName, line)
+			} else {
+				fmt.Printf("%s: invalid %s extractor config: %s\n", rcp.Name, invalidConfigError.PluginName, configError.Message)
+			}
+		}
```

Comment:
This can be refactored into a function to be used by other plugin types (sink, processor)